### PR TITLE
[added] display of server/cluster name to  server report cons/accounts

### PR DESCRIPTION
**Pull request by matthiashanel**
When -j (json output) is provided server info will be included.
For accounts, we collect all server/cluster the account is present in.

Signed-off-by: Matthias Hanel <mh@synadia.com>

```
> nats --context=sys server report connections 9
╭───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                      1 Connections Overview                                                                       │
├──────┬──────────────────────────────┬──────────┬─────────────────┬───────────┬────────────────────────┬────────┬─────────┬──────────┬──────────┬───────────┬──────┤
│ CID  │ Name                         │ Server   │ Cluster         │ IP        │ Account                │ Uptime │ In Msgs │ Out Msgs │ In Bytes │ Out Bytes │ Subs │
├──────┼──────────────────────────────┼──────────┼─────────────────┼───────────┼────────────────────────┼────────┼─────────┼──────────┼──────────┼───────────┼──────┤
│ 5943 │ NATS CLI Version development │ srv-4272 │ cluster-spoke-2 │ 127.0.0.1 │ ABDGQHM46B..Z2QAFGJQKO │ 0s     │ 0       │ 0        │ 0 B      │ 0 B       │ 1    │
╰──────┴──────────────────────────────┴──────────┴─────────────────┴───────────┴────────────────────────┴────────┴─────────┴──────────┴──────────┴───────────┴──────╯
>nats --context=sys server report connections 9 -j
[
  {
    "cid": 6021,
    "ip": "127.0.0.1",
    "port": 59462,
    "start": "2021-05-17T17:42:19.674516Z",
    "last_activity": "2021-05-17T17:42:19.67516Z",
    "rtt": "644µs",
    "uptime": "0s",
    "idle": "0s",
    "pending_bytes": 0,
    "in_msgs": 0,
    "out_msgs": 0,
    "in_bytes": 0,
    "out_bytes": 0,
    "subscriptions": 1,
    "name": "NATS CLI Version development",
    "lang": "go",
    "version": "1.11.0",
    "authorized_user": "UDPCESKLOEOHPSPVD5B2NR3SIXAQNHOBW7GKTEDJNNOJ4NRDUIA6S3RW",
    "account": "ABDGQHM46B4NBP4YIGKBDASU53TZHWPUVI33H5O3RQUPIGZ2QAFGJQKO",
    "subscriptions_list": [
      "_INBOX.0jysFQXmZoVONnNJOoO2hT"
    ],
    "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJlZDI1NTE5LW5rZXkifQ.eyJqdGkiOiJaTFdZRkNaVjQ0Q0lWSDRCR0lHUEhFVUI2S1pOR0ZYU0JIRkZENzZCSjI0Wk9HTzRaQTJRIiwiaWF0IjoxNjE3MzgzNzI2LCJpc3MiOiJBQ1VLNDVVT1dRVTNTWE5CQ1c3RjRYRzdGVElKVUhFM0kyUEozQUY2UUMyQ1NCVFpBTEZWWFZBRCIsIm5hbWUiOiJzeXMiLCJzdWIiOiJVRFBDRVNLTE9FT0hQU1BWRDVCMk5SM1NJWEFRTkhPQlc3R0tURURKTk5PSjROUkRVSUE2UzNSVyIsIm5hdHMiOnsicHViIjp7fSwic3ViIjp7fSwic3VicyI6LTEsImRhdGEiOi0xLCJwYXlsb2FkIjotMSwiaXNzdWVyX2FjY291bnQiOiJBQkRHUUhNNDZCNE5CUDRZSUdLQkRBU1U1M1RaSFdQVVZJMzNINU8zUlFVUElHWjJRQUZHSlFLTyIsInR5cGUiOiJ1c2VyIiwidmVyc2lvbiI6Mn19.pwmB9A0uczAgCYtzEQlYAu2LT6etcwARLgpvJJZCxL0y7yhAFqEamQlbolYSTBjFIlglztMyqMB9e16tZlPeCw",
    "issuer_key": "ACUK45UOWQU3SXNBCW7F4XG7FTIJUHE3I2PJ3AF6QC2CSBTZALFVXVAD",
    "name_tag": "sys",
    "server": {
      "name": "srv-4242",
      "host": "localhost",
      "id": "NBTJKSWVUPPQJ67IPS7LT4ACM2FM7CE2VJMKZSMY7I2PVP3SNJGQMA4T",
      "cluster": "cluster-spoke-1",
      "ver": "2.2.4",
      "seq": 49401,
      "jetstream": true,
      "time": "2021-05-17T17:42:19.676029Z"
    }
  }
]
> nats --context=sys server report accounts
╭──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                               1 Accounts Overview                                                                │
├──────────────────────────────────────────────────────────┬──────────────────────┬─────────────┬─────────┬──────────┬──────────┬───────────┬──────┤
│ Account                                                  │ Server/Cluster       │ Connections │ In Msgs │ Out Msgs │ In Bytes │ Out Bytes │ Subs │
├──────────────────────────────────────────────────────────┼──────────────────────┼─────────────┼─────────┼──────────┼──────────┼───────────┼──────┤
│ ABDGQHM46B4NBP4YIGKBDASU53TZHWPUVI33H5O3RQUPIGZ2QAFGJQKO │ srv-4282/cluster-hub │ 1           │ 0       │ 0        │ 0 B      │ 0 B       │ 1    │
╰──────────────────────────────────────────────────────────┴──────────────────────┴─────────────┴─────────┴──────────┴──────────┴───────────┴──────╯
> nats --context=sys server report accounts -j
[
  {
    "account": "ABDGQHM46B4NBP4YIGKBDASU53TZHWPUVI33H5O3RQUPIGZ2QAFGJQKO",
    "connections": 1,
    "connection_info": [
      {
        "cid": 6125,
        "ip": "127.0.0.1",
        "port": 59600,
        "start": "2021-05-17T17:42:35.142297Z",
        "last_activity": "2021-05-17T17:42:35.142923Z",
        "rtt": "626µs",
        "uptime": "0s",
        "idle": "0s",
        "pending_bytes": 0,
        "in_msgs": 0,
        "out_msgs": 0,
        "in_bytes": 0,
        "out_bytes": 0,
        "subscriptions": 1,
        "name": "NATS CLI Version development",
        "lang": "go",
        "version": "1.11.0",
        "authorized_user": "UDPCESKLOEOHPSPVD5B2NR3SIXAQNHOBW7GKTEDJNNOJ4NRDUIA6S3RW",
        "account": "ABDGQHM46B4NBP4YIGKBDASU53TZHWPUVI33H5O3RQUPIGZ2QAFGJQKO",
        "subscriptions_list": [
          "_INBOX.5R3gElxi9kd2z6tcILDpZC"
        ],
        "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJlZDI1NTE5LW5rZXkifQ.eyJqdGkiOiJaTFdZRkNaVjQ0Q0lWSDRCR0lHUEhFVUI2S1pOR0ZYU0JIRkZENzZCSjI0Wk9HTzRaQTJRIiwiaWF0IjoxNjE3MzgzNzI2LCJpc3MiOiJBQ1VLNDVVT1dRVTNTWE5CQ1c3RjRYRzdGVElKVUhFM0kyUEozQUY2UUMyQ1NCVFpBTEZWWFZBRCIsIm5hbWUiOiJzeXMiLCJzdWIiOiJVRFBDRVNLTE9FT0hQU1BWRDVCMk5SM1NJWEFRTkhPQlc3R0tURURKTk5PSjROUkRVSUE2UzNSVyIsIm5hdHMiOnsicHViIjp7fSwic3ViIjp7fSwic3VicyI6LTEsImRhdGEiOi0xLCJwYXlsb2FkIjotMSwiaXNzdWVyX2FjY291bnQiOiJBQkRHUUhNNDZCNE5CUDRZSUdLQkRBU1U1M1RaSFdQVVZJMzNINU8zUlFVUElHWjJRQUZHSlFLTyIsInR5cGUiOiJ1c2VyIiwidmVyc2lvbiI6Mn19.pwmB9A0uczAgCYtzEQlYAu2LT6etcwARLgpvJJZCxL0y7yhAFqEamQlbolYSTBjFIlglztMyqMB9e16tZlPeCw",
        "issuer_key": "ACUK45UOWQU3SXNBCW7F4XG7FTIJUHE3I2PJ3AF6QC2CSBTZALFVXVAD",
        "name_tag": "sys",
        "server": {
          "name": "srv-4232",
          "host": "localhost",
          "id": "NCIL7YP7NTKKQK53V6KH27GUA3K3IWQE77NRLI556TE52KNQCM6JLOEV",
          "cluster": "cluster-hub",
          "ver": "2.2.4",
          "seq": 51042,
          "jetstream": true,
          "time": "2021-05-17T17:42:35.143801Z"
        }
      }
    ],
    "in_msgs": 0,
    "out_msgs": 0,
    "in_bytes": 0,
    "out_bytes": 0,
    "subscriptions": 1,
    "server": [
      {
        "name": "srv-4232",
        "host": "localhost",
        "id": "NCIL7YP7NTKKQK53V6KH27GUA3K3IWQE77NRLI556TE52KNQCM6JLOEV",
        "cluster": "cluster-hub",
        "ver": "2.2.4",
        "seq": 51042,
        "jetstream": true,
        "time": "2021-05-17T17:42:35.143801Z"
      }
    ]
  }
]
>
```

### CODE REVIEW

File: nats/server_report_command.go [hunk 1]
```diff
@@ -339,18 +341,23 @@ func (c *SrvReportCmd) reportAccount(_ *kingpin.ParseContext) error {
...
 
 	table := newTableWriter(fmt.Sprintf("%d Accounts Overview", len(accounts)))
-	table.AddHeaders("Account", "Connections", "In Msgs", "Out Msgs", "In Bytes", "Out Bytes", "Subs")
+	table.AddHeaders("Account", "Server/Cluster", "Connections", "In Msgs", "Out Msgs", "In Bytes", "Out Bytes", "Subs")
 
 	for _, acct := range accounts {
-		table.AddRow(acct.Account, humanize.Comma(int64(acct.Connections)), humanize.Comma(acct.InMsgs), humanize.Comma(acct.OutMsgs), humanize.IBytes(uint64(acct.InBytes)), humanize.IBytes(uint64(acct.OutBytes)), humanize.Comma(int64(acct.Subs)))
+		server := []string{}
+		for _, s := range acct.Server {
+			server = append(server, fmt.Sprintf("%s/%s", s.Name, s.Cluster))
```
Comment (ripienaar) [inline nats/server_report_command.go:349]: Is cluster always set? I think we generate it right so probably is

File: nats/server_report_command.go [hunk 2]
```diff
@@ -339,18 +341,23 @@ func (c *SrvReportCmd) reportAccount(_ *kingpin.ParseContext) error {
...
+		}
+
+		table.AddRow(acct.Account, strings.Trim(fmt.Sprintf("%v", server), "[]"), humanize.Comma(int64(acct.Connections)), humanize.Comma(acct.InMsgs), humanize.Comma(acct.OutMsgs), humanize.IBytes(uint64(acct.InBytes)), humanize.IBytes(uint64(acct.OutBytes)), humanize.Comma(int64(acct.Subs)))
```
Comment (ripienaar) [inline nats/server_report_command.go:352]: rather use `strings.Join`

File: nats/server_report_command.go [hunk 3]
```diff
@@ -472,31 +492,92 @@ func (c *SrvReportCmd) renderConnections(report []*server.ConnInfo) {
...
+
+	c := connz{
+		Connz:      &server.Connz{},
+		ServerInfo: &server.ServerInfo{},
+	}
+
+	s, ok := reqresp["server"]
+	if !ok {
+		return nil, fmt.Errorf("no server data received in response: %#v", reqresp)
+	}
```
Comment (ripienaar) [inline nats/server_report_command.go:554]: Have to say I find the way these data structures are designed so user unfriendly that they are nearly unusable

File: nats/server_report_command.go [hunk 4]
```diff
@@ -472,31 +492,92 @@ func (c *SrvReportCmd) renderConnections(report []*server.ConnInfo) {
...
-		table.AddRow("", "", "", "", "", humanize.Comma(iMsgs), humanize.Comma(oMsgs), humanize.IBytes(uint64(iBytes)), humanize.IBytes(uint64(oBytes)), humanize.Comma(int64(subs)))
+		table.AddRow("", "", "", "", "", "", "", humanize.Comma(iMsgs), humanize.Comma(oMsgs), humanize.IBytes(uint64(iBytes)), humanize.IBytes(uint64(oBytes)), humanize.Comma(int64(subs)))
 	}
 
 	fmt.Print(table.Render())
 }
 
+type connz struct {
+	*server.Connz
+	*server.ServerInfo
```
Comment (ripienaar) [inline nats/server_report_command.go:513]: Prefer `Connz *server.Connz` I know its nearly the same but prefer that syntax

File: nats/server_report_command.go [hunk 5]
```diff
@@ -472,31 +492,92 @@ func (c *SrvReportCmd) renderConnections(report []*server.ConnInfo) {
...
+	var conns []connInfo
+	for _, conn := range c {
+		for _, c := range conn.Conns {
+			conns = append(conns, connInfo{c, conn.ServerInfo})
+		}
+	}
+	return conns
+}
+
+func parseConnzResp(resp []byte) (*connz, error) {
```
Comment (ripienaar) [inline nats/server_report_command.go:528]: Why return a pointer if you never use it as a pointer?

File: nats/server_report_command.go (hunk 1 above)
Comment (matthiashanel) [reply nats/server_report_command.go:349]: fixed

File: nats/server_report_command.go (hunk 4 above)
Comment (matthiashanel) [reply nats/server_report_command.go:513]: fixed

File: nats/server_report_command.go (hunk 2 above)
Comment (matthiashanel) [reply nats/server_report_command.go:352]: fixed

File: nats/server_report_command.go (hunk 5 above)
Comment (matthiashanel) [reply nats/server_report_command.go:528]: removed the pointer

**Comment by matthiashanel** [comment]
I am leaving connInfo as is as this allows me to only add the server and not change the json any other way.

**Comment by ripienaar** [comment]
LGTM

**Pull request merged**